app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/SyncAccountsWithConnectorExtensionUseCase.kt

```diff
@@ -0,0 +1,119 @@
+@file:Suppress("LongParameterList")
+
+package com.babylon.wallet.android.domain.usecases.p2plink
+
+import com.babylon.wallet.android.data.dapp.PeerdroidClient
+import com.babylon.wallet.android.data.dapp.model.Account
+import com.babylon.wallet.android.data.dapp.model.ConnectorExtensionExchangeInteraction
+import com.babylon.wallet.android.data.repository.p2plink.P2PLinksRepository
+import com.babylon.wallet.android.di.coroutines.IoDispatcher
+import com.radixdlt.sargon.extensions.hash
+import com.radixdlt.sargon.extensions.hex
+import com.radixdlt.sargon.extensions.hexToBagOfBytes
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import rdx.works.core.decodeHex
+import rdx.works.core.hash
+import rdx.works.core.preferences.PreferencesManager
+import rdx.works.core.toHexString
+import rdx.works.peerdroid.data.PeerdroidConnector
+import rdx.works.peerdroid.domain.PeerConnectionStatus
+import rdx.works.profile.data.model.apppreferences.P2PLinkPurpose
+import rdx.works.profile.domain.GetProfileUseCase
+import rdx.works.profile.domain.activeAccountsOnCurrentNetwork
+import timber.log.Timber
+import javax.inject.Inject
+
+class SyncAccountsWithConnectorExtensionUseCase @Inject constructor(
+    private val p2pLinksRepository: P2PLinksRepository,
+    private val getProfileUseCase: GetProfileUseCase,
+    private val peerdroidConnector: PeerdroidConnector,
+    private val peerdroidClient: PeerdroidClient,
+    private val preferencesManager: PreferencesManager,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
```

Comment:
the idea is to avoid `CoroutineDispatcher`s in use cases and keep them in clients/repositories. 
Both the PeerdroidConnector and DataStore are using `ioDispatcher`, so I think you can get rid of them?

Reply:
I've added it for preferences out of habit, there is no actual need for it here since we're using DataStore. Thanks!